tests/Doctrine/Tests/DBAL/Schema/ComparatorTest.php

```diff
@@ -1302,4 +1302,32 @@ public function testForeignKeyRemovalWithRenamedLocalColumn()
         self::assertCount(1, $actual->changedTables['table2']->addedForeignKeys, 'FK to table3 should be added.');
         self::assertEquals('table3', $actual->changedTables['table2']->addedForeignKeys[0]->getForeignTableName());
     }
+
+    public function testAutoincrementColumnWithStringConfiguration()
+    {
+        $schema1 = new Schema([
+            'bugdb' => new Table(
+                'bugdb',
+                [
+                    'integerfield1' => new Column('integerfield1', Type::getType('integer'), [
+                        'autoincrement' => true,
+                    ]),
+                ]
+            ),
+        ]);
+        $schema2 = new Schema([
+            'bugdb' => new Table(
+                'bugdb',
+                [
+                    'integerfield1' => new Column('integerfield1', Type::getType('integer'), [
+                        'autoincrement' => 'auto',
```

Comment:
It should be an integration test. The comparator itself shouldn't account for all potential inconsistencies in platform and schema manager implementations. Does `"auto"` have any different meaning than `true` for MySQL? If not, it should be solved in the schema manager level.